unittest/gunit/keyring_vault/vault_io-t.cc

```diff
@@ -0,0 +1,617 @@
+#include <my_global.h>
+#include <gtest/gtest.h>
+#include <fstream>
+#include "mock_logger.h"
+#include "vault_io.h"
+#include <string.h>
+#include <curl/curl.h>
+
+#if defined(HAVE_PSI_INTERFACE)
+namespace keyring
+{
+  PSI_memory_key key_memory_KEYRING = PSI_NOT_INSTRUMENTED;
+}
+#endif
+
+namespace keyring__vault_io_unittest
+{
+  using namespace keyring;
+
+  using ::testing::Return;
+  using ::testing::StrEq;
+  using ::testing::_;
+  using ::testing::SetArgPointee;
+
+  class Vault_io_test : public ::testing::Test
+  {
+  protected:
+    virtual void SetUp()
+    {
+      credential_file_url = "./keyring_vault.conf";
+      logger= new Mock_logger();
+      vault_curl = new Vault_curl(logger);
+      vault_parser = new Vault_parser(logger);
+    }
+
+    virtual void TearDown()
+    {
+      delete logger;
+    }
+
+  protected:
+    ILogger *logger;
+    IVault_curl *vault_curl;
+    IVault_parser *vault_parser;
+    std::string credential_file_url;
+  };
+
+  TEST_F(Vault_io_test, InitWithNotExisitingCredentialFile)
+  {
+    std::string credential_file_name("./some_funny_name");
+    Vault_io vault_io(logger, vault_curl, vault_parser);
+    remove(credential_file_name.c_str());
+    EXPECT_CALL(*((Mock_logger *)logger),
+      log(MY_ERROR_LEVEL, StrEq("Could not open file with credentials.")));
+    EXPECT_EQ(vault_io.init(&credential_file_name), TRUE);
+
+    remove(credential_file_name.c_str());
+  }
+
+  TEST_F(Vault_io_test, InitWithInvalidToken)
+  {
+    Vault_io vault_io(logger, vault_curl, vault_parser);
+    std::string conf_with_invalid_token("invalid_token.conf");
+
+    std::remove(conf_with_invalid_token.c_str());
+    std::ofstream my_file;
+    my_file.open(conf_with_invalid_token.c_str());
+    my_file << "vault_url = https://127.0.0.1:8200" << std::endl;
+    my_file << "secret_mount_point = secret" << std::endl;
+    my_file << "token = 123-123-123" << std::endl;
+    my_file << "vault_ca = ./vault_ca.crt";
+    my_file.close();
+
+    EXPECT_EQ(vault_io.init(&conf_with_invalid_token), FALSE);
+
+    EXPECT_CALL(*((Mock_logger *)logger),
+      log(MY_ERROR_LEVEL, StrEq("Could not retrieve list of keys from Vault. "
+          "Vault has returned the following error(s): [\"permission denied\"]")));
+    ISerialized_object *serialized_keys= NULL;
+    EXPECT_EQ(vault_io.get_serialized_object(&serialized_keys), TRUE);
+
+    std::remove(conf_with_invalid_token.c_str());
+  }
+
+/*
```

Comment:
The tests are checking what happens when incorrect path to CA is given and if no CA is given at all. Those test make sense (very much indeed), but only if the machine on which test suite is run do not trust CA that issued certificate for Vault server. In this case keyring_vault (or CURL to be specific) will use CA certificate provided by user in configuration file (vault_ca option) to verify Vault's server certificate.